packages/mux-player/src/index.ts

```diff
@@ -0,0 +1,473 @@
+    // But just fixing it manually for now.
+    if (el.video) {
+      el.video.muted = el.video.defaultMuted;
+    }
+
+    if (el.video?.hls) {
+      // Temporarily here to load less segments on page load, remove later!!!!
+      el.video.hls.config.maxMaxBufferLength = 2;
+    }
+
+    this._setUpErrors(el);
+    this._setUpMutedAutoplay(el);
+    this._setUpCaptionsButton(el);
+    this._setUpAirplayButton(el);
+    this._setUpVolumeRange(el);
+  }
+
+  connectedCallback() {
+    this._renderChrome();
+    this._initResizing();
+  }
+
+  disconnectedCallback() {
+    this._deinitResizing();
+  }
+
+  get _fragments() {
+    return this._template.fragments;
+  }
+
+  _renderChrome() {
+    if (this._playerSize != getPlayerSize(this.el)) {
+      this._playerSize = getPlayerSize(this.el);
+      this._fragments.chromeRenderer.render(getProps(this.el, this._state));
+    }
+  }
+
+  _initResizing() {
+    this._resizeObserver = new ResizeObserver(() => this._renderChrome());
+    this._resizeObserver.observe(this.el);
+  }
+
+  _deinitResizing() {
+    this._resizeObserver?.disconnect();
+  }
+
+  _setUpErrors(el: MuxPlayerElement) {
+    const onError = (error: { code: number } | null | undefined) => {
+      // Don't show an error dialog on an abort error.
+      if (!error?.code || error.code === MediaError.MEDIA_ERR_ABORTED) {
+        return;
+      }
+
+      switch (error.code) {
+        case MediaError.MEDIA_ERR_NETWORK:
+          this._state.dialog = {
+            title: "Network Error",
+            message:
+              "A network error occurred. Please reload the player and try again.",
+          };
+          break;
+        case MediaError.MEDIA_ERR_DECODE:
+          this._state.dialog = {
+            title: "Media Error",
+            message:
+              "A media error occurred. Please reload the player and try again.",
+          };
+          break;
+        case MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED:
+          this._state.dialog = {
+            title: "Source Not Supported",
+            message: "This media source is not supported.",
+          };
+          break;
+        default:
+          this._state.dialog = {
+            title: "Error",
+            message:
+              "An error occurred. Please reload the player and try again.",
+          };
+          break;
+      }
+      this._fragments.dialogContent.render(this._state.dialog);
+      (this._fragments.dialogContent.parentNode as MxpDialog)?.show();
+    };
+
+    if (el.video?.hls) {
+      const Hls: any = el.video.hls.constructor;
+      const onHlsError = (_event: any, data: any) => {
+        if (data.fatal) {
+          const errorMap = {
+            [Hls.ErrorTypes.NETWORK_ERROR]: MediaError.MEDIA_ERR_NETWORK,
+            [Hls.ErrorTypes.MEDIA_ERROR]: MediaError.MEDIA_ERR_DECODE,
+          };
+          onError({ code: errorMap[data.type] });
+        }
+      };
+      el.video.hls.on(Hls.Events.ERROR, onHlsError);
```

Comment:
blocking: we need mux-player to redispatch an error event of some sort for these errors (stream.new relies on this)